### DISCUSSION TITLE
Is there a way to get a translation by string like in non typesafe i18n tools?

### DISCUSSION THREAD

**Discussion by m1212e** [ivanhofer/typesafe-i18n#126] [Q&A] [answered]
Hi,

I know this would kill the whole purpose of the typesafe thing, but is there a way to retrieve translations by a . separated string?
I'm migrating an older app to this new i18n tool and in most cases I just need to replace the string with the according call on the LL object. But in some cases the strings are generated by some logic (e.g. the months of the year are translated by calling `'months.' + item.month()`) and therefore cannot be hard coded. Is there an easy way to access those translations dynamically? Did I just overlook that in the docs?
Currently I'm just doing it this way: `LL.months[item.month()]()`
Is this the correct way to access those cases?
Thanks in advance!

**Comment by ivanhofer** [comment, marked as answer]
Hi, currently it is only possible to access the translations via an object-syntax. So each part has to be split on the '.' character. For dynamic attributes like the month, you need to use the `[]` syntax.
Your example looks right 👍
With dynamic values you could probably get type-errors so you will need to cast it. If `item. month()` returns a `number` but your translation will only contain values ranging from `1 - 12`you will need to write it like this: `LL.months[item.months() as any]()`.

Hope this answers your question.

**Reply by m1212e** [reply]
It does, thanks a lot. Since there are some legacy use cases inside the app I'll probably have to find a workaround. Do you plan on adding this feature as a kind of backup for edge cases like this, or to allow compatibility with older, string based translation systems?

**Reply by ivanhofer** [reply]
no sorry, not really. I don't see the demand for an additional way to access the translations. And I can't really account for all possible migrations from other libraries. 
If you are switching from an existing solution you can probably do a regex find and replace on all files and it will cover 90% of your translations already. With TypeScript you then will instantly see all errors and you can fix the remaining errors manually.
If you find a good solution on how to migrate, please document it here. Maybe I will add a new FAQ section. to the Readme file.

**Reply by m1212e** [reply]
Yeah thats what I did. Worked pretty ok. For some cases I need to find a workaround but thats ok. Thanks for your time!

**Reply by ivanhofer** [reply]
@m1212e gappy to see that you have found a solution :)

A small sidenote:
I just tested a few things and the resolving of `LL['months.' + item.month()]()`, like you mentioned in the original message actually works.
You would get a TypeScript error when writing it like this, but when running the code, it will work,
Nevertheless you should still use the official supported way and call it with the object syntax `LL.months[item.months()]()`
